exporter/clickhouseexporter/internal/exponential_histogram_metrics.go

```diff
@@ -117,54 +113,55 @@ func (e *expHistogramMetrics) insert(ctx context.Context, db *sql.DB) error {
 		return nil
 	}
 
-	valueArgs := make([]any, e.count*expHistogramValueCounts)
-	var b strings.Builder
-
-	index := 0
-	for _, model := range e.expHistogramModels {
-		for i := 0; i < model.expHistogram.DataPoints().Len(); i++ {
-			dp := model.expHistogram.DataPoints().At(i)
-			b.WriteString(*expHistogramPlaceholders)
-
-			valueArgs[index] = model.metadata.ResAttr
-			valueArgs[index+1] = model.metadata.ResURL
-			valueArgs[index+2] = model.metadata.ScopeInstr.Name()
-			valueArgs[index+3] = model.metadata.ScopeInstr.Version()
-			valueArgs[index+4] = attributesToMap(model.metadata.ScopeInstr.Attributes())
-			valueArgs[index+5] = model.metadata.ScopeInstr.DroppedAttributesCount()
-			valueArgs[index+6] = model.metadata.ScopeURL
-			valueArgs[index+7] = model.metricName
-			valueArgs[index+8] = model.metricDescription
-			valueArgs[index+9] = model.metricUnit
-			valueArgs[index+10] = attributesToMap(dp.Attributes())
-			valueArgs[index+11] = dp.StartTimestamp().AsTime().UnixNano()
-			valueArgs[index+12] = dp.Timestamp().AsTime().UnixNano()
-			valueArgs[index+13] = dp.Count()
-			valueArgs[index+14] = dp.Sum()
-			valueArgs[index+15] = dp.Scale()
-			valueArgs[index+16] = dp.ZeroCount()
-			valueArgs[index+17] = dp.Positive().Offset()
-			valueArgs[index+18] = convertSliceToArraySet(dp.Positive().BucketCounts().AsRaw())
-			valueArgs[index+19] = dp.Negative().Offset()
-			valueArgs[index+20] = convertSliceToArraySet(dp.Negative().BucketCounts().AsRaw())
-
-			attrs, times, values, traceIDs, spanIDs := convertExemplars(dp.Exemplars())
-			valueArgs[index+21] = attrs
-			valueArgs[index+22] = times
-			valueArgs[index+23] = values
-			valueArgs[index+24] = traceIDs
-			valueArgs[index+25] = spanIDs
-			valueArgs[index+26] = uint32(dp.Flags())
-			valueArgs[index+27] = dp.Min()
-			valueArgs[index+28] = dp.Max()
-
-			index += expHistogramValueCounts
-		}
-	}
-
 	start := time.Now()
 	err := doWithTx(ctx, db, func(tx *sql.Tx) error {
-		_, err := tx.ExecContext(ctx, fmt.Sprintf("%s %s", e.insertSQL, strings.TrimSuffix(b.String(), ",")), valueArgs...)
+		batch, err := tx.PrepareContext(ctx, e.insertSQL)
+		if err != nil {
+			return err
+		}
+
+	batch:
```

Comment:
not recommend use `goto` , pls reference https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/exporter_logs.go#L60